test/e2e/aadpodidentity_test.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/Azure/aad-pod-identity/test/e2e/config"
+	"github.com/Azure/aad-pod-identity/test/e2e/deploy"
+	"github.com/Azure/aad-pod-identity/test/e2e/pod"
+	"github.com/Azure/aad-pod-identity/test/e2e/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	cfg                config.Config
+	templateOutputPath = path.Join("template", "_output")
+)
+
+var _ = BeforeSuite(func() {
+	fmt.Println("Setting up the test suite environment...")
+
+	// Create a folder '_output' in template/ for storing temporary deployment files
+	err := os.Mkdir(templateOutputPath, os.ModePerm)
+	Expect(err).NotTo(HaveOccurred())
+
+	c, err := config.ParseConfig()
+	Expect(err).NotTo(HaveOccurred())
+	cfg = *c // To avoid 'Declared and not used' linting error
+
+	// Install CRDs and deploy MIC and NMI
+	cmd := exec.Command("kubectl", "apply", "-f", "../../deploy/infra/deployment-rbac.yaml")
+	util.PrintCommand(cmd)
+	_, err = cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	fmt.Println("Tearing down the test suite environment...")
+
+	// Uninstall CRDs and delete MIC and NMI
+	cmd := exec.Command("kubectl", "delete", "-f", "../../deploy/infra/deployment-rbac.yaml", "--ignore-not-found")
+	util.PrintCommand(cmd)
+	_, err := cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred())
+
+	err = os.RemoveAll(templateOutputPath)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
+	BeforeEach(func() {
+		fmt.Println("Setting up the test environment...")
+
+		err := azureidentity.CreateOnCluster(cfg.SubscriptionID, cfg.ResourceGroup, "test-identity", templateOutputPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = azureidentitybinding.Create("test-identity", templateOutputPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = deploy.Create(cfg.SubscriptionID, cfg.ResourceGroup, "identity-validator", "test-identity", templateOutputPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		ok, err := deploy.WaitOnReady("identity-validator")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ok).To(Equal(true))
+	})
+
+	AfterEach(func() {
+		fmt.Println("Tearing down the test environment...")
+
+		cmd := exec.Command("kubectl", "delete", "AzureIdentity,AzureIdentityBinding,AzureAssignedIdentity", "--all")
+		util.PrintCommand(cmd)
+		_, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred())
+
+		ok, err := azureassignedidentity.WaitOnDeletion()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ok).To(Equal(true))
+	})
+
+	It("should pass the identity validating test", func() {
+		clientID, err := azureidentity.GetClientID(cfg.ResourceGroup, "test-identity")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(clientID).NotTo(Equal(""))
+
+		list, err := azureassignedidentity.GetAll()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(list.Items).To(HaveLen(1))
+		Expect(list.Items[0].Spec.Pod).NotTo(Equal(""))
```

Comment:
Can you please confirm that azureassignedidentity which has the right properties got created as well ?